### PR TITLE
Allow \RedisArray and \RedisCluster objects for RedisCachePool

### PR DESCRIPTION
**Pull request by wshafer**
| Question      | Answer
| ------------- | ------
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | src/Adapter/Redis/README.md

### Description
Allows a \RedisArray or \RedisCluster to also be used with the RedisCachePool adapter.
Adds to unit tests for each types
Adds cluster setup script to travis for unit and integration tests

### TODO
* [ ] Updated Changelog.md

### CODE REVIEW

**Comment by wshafer** [comment]
Forcing Rebuild

**Comment by wshafer** [comment]
Forcing Rebuild

**Comment by wshafer** [comment]
Let's see if Travis has their act together finally

File: src/Adapter/Redis/Tests/CreateArrayPoolTrait.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Cache\Adapter\Redis\Tests;
+
+use Cache\Adapter\Redis\RedisCachePool;
+
+trait CreateArrayPoolTrait
```
Comment (prisis) [inline src/Adapter/Redis/Tests/CreateArrayPoolTrait.php]: Can you change the traits names to something like this CreateRedisArrayCachePoolTrait

**Comment by wshafer** [comment]
Sorry for the delay.  Updates completed

**Comment by wshafer** [comment]
PS. I think the Style checker is broken.  It's showing changes never made by this pr.  :(

File: src/Adapter/Redis/README.md [hunk 2]
```diff
@@ -20,14 +20,28 @@ composer require cache/redis-adapter
 
 ### Use
 
-To create an instance of `RedisCachePool` you need to configure a `\Redis` client. 
+To create an instance of `RedisCachePool` you need to configure a `\Redis`, `\RedisArray`, or `\RedisCluster` client.
```
Comment (prisis) [inline src/Adapter/Redis/README.md]: remove the comma before the or

File: src/Adapter/Redis/RedisCachePool.php [hunk 3]
```diff
@@ -29,10 +30,19 @@ class RedisCachePool extends AbstractCachePool implements HierarchicalPoolInterf
     protected $cache;
 
     /**
-     * @param \Redis $cache
+     * @param \Redis|\RedisArray|\RedisCluster $cache
      */
-    public function __construct(\Redis $cache)
+    public function __construct($cache)
     {
+        if (!$cache instanceof \Redis
```
Comment (prisis) [inline src/Adapter/Redis/RedisCachePool.php]: shouldn't this check be more like this 

if ($cache instanceof \Redis || $cache instanceof \RedisArray || $cache instanceof \RedisCluster) {
     $this->cache = $cache;
} else {
    throw new CachePoolException('Cache instance must be of type \Redis, \RedisArray or \RedisCluster');
}

**Comment by prisis** [comment]
And ignore the style checker

File: src/Adapter/Redis/RedisCachePool.php (hunk 3 above)
Comment (wshafer) [reply src/Adapter/Redis/RedisCachePool.php]: I can certainly make it like that. But I’d push back a little and say the else clause isn’t needed at all and just muddies the code a bit.  Why execute anything if it doesn’t meet the conditions needed to execute?  However this is not my package, so if you’d like that change I can certainly make it.

Comment (wshafer) [reply src/Adapter/Redis/RedisCachePool.php]: I can certainly make it like that. But I’d push back a little and say the else clause isn’t needed at all and just muddies the code a bit.  Why execute anything if it doesn’t meet the conditions needed to execute?  However this is not my package, so if you’d like that change I can certainly make it.

Comment (wshafer) [reply src/Adapter/Redis/RedisCachePool.php]: Good catch... I'll fix this

Comment (prisis) [reply src/Adapter/Redis/RedisCachePool.php]: Maybe you can add a small test for this

Comment (wshafer) [reply src/Adapter/Redis/RedisCachePool.php]: Test was added.  Thanks agaib

**Comment by prisis** [comment]
Thank you, for your work!

File: src/Adapter/Redis/Tests/RedisCachePoolTest.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace Cache\Adapter\Redis\Tests;
+
+use Cache\Adapter\Redis\RedisCachePool;
+
+class RedisCachePoolTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * Tests that an exception is thrown if invalid object is
+     * passed to the constructor
```
Comment (prisis) [inline src/Adapter/Redis/Tests/RedisCachePoolTest.php]: add a dot after constructor

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# -*- mode: ruby -*-
```
Comment (prisis) [inline Vagrantfile]: can you remove this file

**Comment by prisis** [comment]
Can you rebase this against master so that i can merge this

**Comment by wshafer** [comment]
@prisis - Rebased.  Thanks

**Comment by prisis** [comment]
Thank you, for your work!

**Pull request merged**